crates/rpc/src/routes/mod.rs

```diff
@@ -3,28 +3,35 @@ use std::sync::Arc;
 use beacon::get_beacon_routes;
 use config::get_config_routes;
 use node::get_node_routes;
+use debug::get_debug_routes;
 use ream_network_spec::networks::NetworkSpec;
 use ream_storage::db::ReamDB;
 use warp::{Filter, Rejection, path, reply::Reply};
 
 pub mod beacon;
 pub mod config;
 pub mod node;
+pub mod debug;
 
 /// Creates and returns all possible routes.
 pub fn get_routes(
     network_spec: Arc<NetworkSpec>,
     db: ReamDB,
 ) -> impl Filter<Extract = impl Reply, Error = Rejection> + Clone {
-    let eth_base = path("eth").and(path("v1"));
+    let eth_base_v1 = path("eth").and(path("v1"));
+    let eth_base_v2 = path("eth").and(path("v2"));
 
-    let beacon_routes = get_beacon_routes(network_spec.clone(), db);
+    let beacon_routes = get_beacon_routes(network_spec.clone(), db.clone());
 
     let node_routes = get_node_routes();
 
     let config_routes = get_config_routes(network_spec.clone());
 
-    eth_base.and(beacon_routes.or(node_routes).or(config_routes))
+    let debug_routes = get_debug_routes(db.clone());
+
+    let combined_routes = beacon_routes.or(node_routes).or(config_routes).or(debug_routes);
+    let versioned_base = eth_base_v1.or(eth_base_v2).and_then(|_| async { Ok::<_, Rejection>(()) });
+    versioned_base.and(combined_routes).map(|_, reply| reply)
```

Comment:
v1 routes should only reply to v1 same with v2, wouldn't this current approach let you use either version for every RPC? which is unidle, maybe make a `debug_routes_v2` function, and consider ones without `_v1` v1

Reply:
Yeah, will fix this.